tests/test_instantiate.py

```diff
@@ -25,6 +25,21 @@ def testInstantiate(self):
         mock_args.input_file = None
         mkcc = mkchromecast.Mkchromecast(mock_args)
 
+    def testMP3CodecNodeBackend(self):
+        "This test evaluates the assignment of the MP3 codec when the Node Backend is selected"
+
+        mock_args = mock.Mock()
+        # Here we set the minimal required args for __init__ to not sys.exit.
+        mock_args.encoder_backend = 'node'
+        mock_args.bitrate = constants.DEFAULT_BITRATE
+        mock_args.codec = 'mp3'
+        mock_args.command = None
+        mock_args.resolution = None
+        mock_args.chunk_size = 64
+        mock_args.sample_rate = 44100
+        mock_args.youtube = None
+        mock_args.input_file = None
+        mkcc = mkchromecast.Mkchromecast(mock_args)
     def testTrayModeInstantiation(self):
```

Comment:
Minor comment: there should be a newline between function definitions.

Reply:
Hmm, there should have been a space, nice catch!

Reply:
My recommendation would be to use a formatter like this: https://github.com/psf/black In that way, you don't have to worry about missing newlines, spaces, and all those kinds of stuff.

Reply:
I usually do actually, but I am very used to using poetry, and used pyenv with virtualenvs to make this project work, which put me a little off balance tbh. I am a big fan ruff btw :)

Reply:
Yes, doing some auto-formatting is in the plans.  Auto-formatting in the middle of a refactor is usually pretty disruptive, though, which is why I've been holding off on much formatting at this point.  My plan was to keep focusing on the general organization and health of the codebase, and to think about formatting once things feel stable again.

Reply:
True. I thought I had applied autoformatting before, but I don't know if you did once you refactored.

Reply:
In general, here are the things I'd like to be done before thinking about code formatting:

- Create some parallel structure across cast.py, audio.py, video.py, and node.py.  In particular, get out of the current state where Node support is split between node.py and video.py
- Re-integrate Sonos support
- Avoid having code use `config.Config` directly (they should all go through `Mkchromecast`)
- Avoid using on-disk temporary files for anything
- Keep track of child processes and manage them directly, instead of looping through `psutil` and killing things
- Have a single, centralized way to terminate the application